src/ConstructiveSolidGeometry/plotting/Meshing.jl

```diff
@@ -8,17 +8,16 @@
 (X[i,j], Y[i,j], Z[i,j]) is a cartesian point
 """
 
-struct Mesh{T, N}
-    x::Array{T,N}
-    y::Array{T,N}
-    z::Array{T,N}
-    function Mesh(
-        x::Array{T,N},
-        y::Array{T,N},
-        z::Array{T,N}) where {T,N}
-        @assert size(x) == size(y) && size(x) == size(z)
-        new{T,N}(x, y, z)
-    end
+struct Mesh{T}
+    x::Array{T,2}
+    y::Array{T,2}
+    z::Array{T,2}
+end
+
+struct PolyMesh{T,N}
+    x::Vector{SVector{N,T}}
+    y::Vector{SVector{N,T}}
+    z::Vector{SVector{N,T}}
 end
```

Comment:
Just to clarify: `PolyMesh` is an auxiliary struct for plotting using `mesh3d` in `GR`?

Reply:
For now this it is its only purpose. However, I will need PolyMesh{T,3} for when I plot surfaces for which I cannot write a analytical mesh. These arbitrary meshes will be generated by the JuliaGeometry/VoronoiDelaunay.jl package and then stored in a `PolyMesh{T,3}` (which contains vectors of triangle vertices. `PolyMesh{T,4}` contains vectors of quadrangle vertices). Note that pyplot() understands PolyMesh{T} aswell. So in theory we do not need to check for the backend and can always pass a PolyMesh. But we save time by not doing this in pyplot(). Another reason to use Mesh{T} is that pyplot() looks really nice with this structure.

Reply:
This all is very annoying... I don't mean you implementation but this whole Plots.jl, GR/PyPlot stuff for 3D plots...

[Meshes.jl](https://github.com/JuliaGeometry/Meshes.jl) actually seems to make really nice progress and provides a mesh struct we can use. 
```julia
using Plots
using Meshes

T = Float64
p0 = Point3([0,0,0])
px = Point3([1,0,0])
py = Point3([0,1,0])
pz = Point3([0,0,1])

# mesh of three triangles, just as an example
mesh = SimpleMesh([p0, px, py, pz], [connect((1, 2, 3)), connect((1, 2, 4))])

plot(mesh)
```
This produces a wireframe in GR. With PyPlot it throws an error....
I would favor to use the mesh from Meshes.jl and, as they also provide already plot recipes for Plots.jl 
(even though they plot usually with [MeshViz.jl](https://github.com/JuliaGeometry/MeshViz.jl)), talk to them regarding your workaround for plotting nice surfaces via PyPlot to their package. They suggest to contact them via https://julialang.zulipchat.com/#narrow/stream/275558-meshes.2Ejl before making issues on GitHub. Maybe we could also improve/contribute to Plots.jl directly. 
And we might talk to Josef Heinen regarding face support from GR.jl. 

I know this will increase the time until everything is through, but I think this is the better way and will reduce the amount of code in our package as we would only need to implement the connections for the vertices of our volume primitives to create the meshes.